src/basic/hashmap.c

```diff
@@ -1775,22 +1775,30 @@ int hashmap_put_strdup(Hashmap **h, const char *k, const char *v) {
                 return r;
 
         _cleanup_free_ char *kdup = NULL, *vdup = NULL;
+
         kdup = strdup(k);
-        vdup = strdup(v);
-        if (!kdup || !vdup)
+        if (!kdup)
                 return -ENOMEM;
 
+        if (v) {
+                vdup = strdup(v);
+                if (!vdup)
+                        return -ENOMEM;
+        }
```

Comment:
what's the rationale for this? i mean, hashmap_get() for a NULL entry and for a non-existing entry is non-distinguishable. Is this really something we want? Sounds like trap to me...

Reply:
I was transforming the code in device-enumerator.c, without changing the logic. It has explicit support for specifying a key with a NULL value. See  `sd_device_enumerator_add_match_property` and  `match_property()`. It'll then match properties with a NULL value. I'm not sure how much sense this makes, but its an exported API.